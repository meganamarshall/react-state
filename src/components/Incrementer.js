import React, { PureComponent } from 'react';

export default class Incrementer extends PureComponent {
  state = {
    count: 0
  }

  incrementCount = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  }

  render() {
    const { count } = this.state;
    return (
      <>
        <p>Current Count: {count}</p>
        <button onClick={this.incrementCount}>Increment</button>
      </>
    );
  }
}

