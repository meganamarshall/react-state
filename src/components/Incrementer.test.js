import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('incrementer component', () => {
  it('renders the incrementer', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('increments the count', () => {
    const wrapper = shallow(<Incrementer />);

    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.state('count')).toEqual(1);
  });
});


