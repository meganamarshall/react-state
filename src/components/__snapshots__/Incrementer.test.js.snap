// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incrementer component renders the incrementer 1`] = `
<Fragment>
  <p>
    Current Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</Fragment>
`;
